ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -1947,6 +1947,50 @@ char *MM_oemansi_n(char *szszChain, size_t n_bytes)
     return szszChain;
 }
 
+// An implementation of non-sensitive strstr()
+char *MM_stristr(const char *haystack, const char *needle)
+{
+    if (!haystack)
+        return nullptr;
+
+    if (!needle)
+        return nullptr;
+
+    if (!*needle)
+        return (char *)haystack;
+
+    char *p1 = (char *)haystack;
+    char *p2 = (char *)needle;
+    char *p1Adv = (char *)haystack;
+
+    while (*++p2)
+    {
+        p1Adv++;
+    }
+
```

Comment:
I can't make sense of that. Why would you skip the first strlen(needle) character of haystack ? That's wrong, and that can cause an out-of-bounds access if strlen(haystack) < strlen(needle). And please use EQUALN() to simplify the code

Reply:
Sorry, I've rewritten it again using the EQUALN() function. 
Now it's very more clear.